shared-services/README.md

```diff
@@ -0,0 +1,172 @@
+# DevX 
+
+The `devx` flake provisions a development environment suitable for working on a `haskell.nix` project.
+
+It includes a complete haskell toolchain as well as all the native dependencies needed to work with other haskell IOG repositories.
+
+# Test that `devx` can build your Haskell project 
+
+Pick your `ghc` version (one of `ghc8107`, `ghc901`, `ghc924`), and enter the shell:
+```
+nix develop github:zeme-iohk/devx#ghc8107 --no-write-lock-file
+```
+You may now build and test your project:
+```
+cabal update
+cabal build all -j 
+cabal test all 
+```
+
+If you wish to use `devx` in your project:
+- If you have a new repository then [click here](#integration-with-a-new-repository) for instructions.
+- If you wish to migrate your existing nix code then [click here](#integration-with-an-existing-repository) instead.
+
+# Integration with a new repository 
+
+If this is a new repository, you should use the flake template.
+
+```
+nix flake init --template github:zeme-iohk/devx
+```
+
+Two files will be created in the top-level: `flake.nix` and `devx.nix`.
+
+[Click here](#the-devx-flakenix-file) to continue learning.
+
+# Integration with an existing repository
+
+In order to successully migrate to `devx` your nix code will have to be refactored until your `flake.nix` is standardized (see [here](#the-standard-flakenix-file)) and your `devx.nix` file (which you have to create) becomes the entrypoint to all your nix code (see [here](#the-devxnix-file)).
+
+# The standard `flake.nix` file
+
+Projects using `devx` share an almost identical `flake.nix`:
+```nix
+{
+  description = "";
+
+  inputs.devx.url = "github:zeme-iohk/devx";
+
+  # You can add other flake inputs here if you must: inputs.other = {}
+
+  outputs = inputs.devx.lib.mkFlakeOutputs;
+
+  nixConfig = {
+    extra-substituters = [
+      "https://cache.iog.io"
+      "https://cache.zw3rk.com"
+    ];
+    extra-trusted-public-keys = [
+      "hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ="
+      "loony-tools:pr9m4BkM/5/eSTZlkQyRt57Jz7OMBxNSUiMC4FkcNfk="
+    ];
+    accept-flake-config = true;
+    allow-import-from-derivation = true;
+  };
+}
+```
+You may edit this file in three ways:
+1. Fill the `description` to your liking.
+2. Edit the `nixConfig` attrset, but do not **no** invalidate the cache-related fields.
```

Comment:
Should describe which kinds of changes to the `nixConfig` we expect users to commonly want to think about (if any), as well as links to appropriate official docs.

Instead of "don't invalidate" I'd say "do not remove any elements from extra-subs and extra-keys"